src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -1,23 +1,31 @@
-package org.broadinstitute.dsde.workbench.sam.service
+package org.broadinstitute.dsde.workbench.sam
+package service
 
+import akka.http.scaladsl.model.StatusCodes
 import cats.effect.IO
 import com.typesafe.scalalogging.LazyLogging
-import org.broadinstitute.dsde.workbench.model.{WorkbenchEmail, WorkbenchGroupName}
+
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchEmail, WorkbenchExceptionWithErrorReport, WorkbenchGroupName, WorkbenchSubject}
 import org.broadinstitute.dsde.workbench.sam.dataAccess.DirectoryDAO
 import org.broadinstitute.dsde.workbench.sam.model.BasicWorkbenchGroup
 import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+import org.broadinstitute.dsde.workbench.sam.config.TermsOfServiceConfig
+
+import scala.concurrent.ExecutionContext
+
+//import scala.concurrent.ExecutionContext
 
-class TosService (val directoryDao: DirectoryDAO, val appsDomain: String) extends LazyLogging {
 
-  def createNewGroupIfNeeded(currentVersion: Int, isEnabled: Boolean): IO[Option[BasicWorkbenchGroup]] = {
-    if(isEnabled) {
-      getTosGroup(currentVersion).flatMap {
-        case Some(_) =>
-          IO.none
+class TosService (val directoryDao: DirectoryDAO, val appsDomain: String, val tosConfig: TermsOfServiceConfig)(implicit val executionContext: ExecutionContext) extends LazyLogging {
+
+  def createNewGroupIfNeeded(): IO[Option[BasicWorkbenchGroup]] = {
+    if(tosConfig.enabled) {
+      getTosGroup().flatMap {
         case None =>
```

Comment:
Will we only get NONE if the group doesn't exist yet? What happens if we get another type of exception, like a networking error, when trying to look up the group? Is this returned as an exception?

Reply:
I think this only returns None if the group does not exist yet